generated/nidaqmx/_lib.py

```diff
@@ -148,39 +151,72 @@ def task_handle(self) -> type:
         self._cdll = None
+        self._encoding = None
 
         windll = None
         cdll = None
-
-        if sys.platform.startswith('win') or sys.platform.startswith('cli'):
-            try:
-                if 'iron' in platform.python_implementation().lower():
-                    windll = ctypes.windll.nicaiu
-                    cdll = ctypes.cdll.nicaiu
-                else:
-                    windll = ctypes.windll.LoadLibrary('nicaiu')
-                    cdll = ctypes.cdll.LoadLibrary('nicaiu')
-            except (OSError, WindowsError) as e:
-                raise DaqNotFoundError(_DAQ_NOT_FOUND_MESSAGE) from e
+        encoding = None
+
+        # Feature Toggle to load nicaiu.dll or nicai_utf8.dll
+        # The Feature Toggle can be set in the .env file
+        nidaqmx_c_library = config('NIDAQMX_C_LIBRARY', cast=str, default=None)
+
+        if sys.platform.startswith('win') or sys.platform.startswith('cli'):   
+            if nidaqmx_c_library is not None:
+                try: 
+                    if nidaqmx_c_library=="nicaiu":
+                        windll, cdll = _load_lib("nicaiu")
+                        encoding = locale.getlocale()[1]
+                    elif nidaqmx_c_library=="nicai_utf8":
+                        windll, cdll = _load_lib("nicai_utf8")
+                        encoding = 'utf-8'  
+                    else:
+                        raise DaqNotFoundError(_DAQ_NOT_FOUND_MESSAGE)
+                except (OSError, WindowsError) as e:
+                    raise DaqNotFoundError(_DAQ_NOT_FOUND_MESSAGE) from e         
+            else:
+                try:
+                    windll, cdll = _load_lib("nicai_utf8")
+                    encoding = 'utf-8'  
+                except (OSError, WindowsError):
+                    # Fallback to nicaiu.dll if nicai_utf8.dll cannot be loaded
+                    try:
+                        windll, cdll = _load_lib("nicaiu")
+                        encoding = locale.getlocale()[1]
+                    except (OSError, WindowsError) as e:
+                        raise DaqNotFoundError(_DAQ_NOT_FOUND_MESSAGE) from e         
         elif sys.platform.startswith('linux'):
             # On linux you can use the command find_library('nidaqmx')
             if find_library('nidaqmx') is not None:
                 cdll = ctypes.cdll.LoadLibrary(find_library('nidaqmx'))
                 windll = cdll
+                encoding = locale.getlocale()[1]
             else:
                 raise DaqNotFoundError(_DAQ_NOT_FOUND_MESSAGE)
         else:
             raise DaqNotSupportedError(_DAQ_NOT_SUPPORTED_MESSAGE.format(sys.platform))
 
         self._windll = DaqFunctionImporter(windll)
         self._cdll = DaqFunctionImporter(cdll)
+        self._encoding = encoding
+
+        def _load_lib(libname: str):
+            windll = ctypes.windll.LoadLibrary(libname)
+            cdll = ctypes.cdll.LoadLibrary(libname)
+            return windll, cdll
```

Comment:
This is only used inside the `if sys.platform.startswith('win')` block, so please move it there.